WalletWasabi.Fluent/Models/Transactions/PrivacySuggestionsModel.cs

```diff
@@ -44,30 +46,31 @@ public PrivacySuggestionsModel(Wallet wallet)
 	/// <remarks>Method supports being called multiple times. In that case the last call cancels the previous one.</remarks>
 	public async Task<PrivacySuggestionsResult> BuildPrivacySuggestionsAsync(TransactionInfo info, BuildTransactionResult transactionResult, CancellationToken cancellationToken)
 	{
-		var result = new PrivacySuggestionsResult();
+		var result = PrivacySuggestionsResult.CreateNew();
 
 		using CancellationTokenSource singleRunCts = new();
 
 		lock (_lock)
 		{
-			_suggestionCancellationTokenSource?.Cancel();
-			_suggestionCancellationTokenSource = singleRunCts;
+			_singleRunCancellationTokenSource?.Cancel();
+			_linkedCancellationTokenSource?.Cancel();
+			_singleRunCancellationTokenSource = singleRunCts;
+			CancellationTokenSource timeoutCts = new(TimeSpan.FromSeconds(15));
+			CancellationTokenSource linkedCts = CancellationTokenSource.CreateLinkedTokenSource(timeoutCts.Token, singleRunCts.Token, cancellationToken);
+			_linkedCancellationTokenSource = linkedCts;
 		}
 
-		using CancellationTokenSource timeoutCts = new(TimeSpan.FromSeconds(15));
-		using CancellationTokenSource linkedCts = CancellationTokenSource.CreateLinkedTokenSource(timeoutCts.Token, singleRunCts.Token, cancellationToken);
-
 		using (await _asyncLock.LockAsync(CancellationToken.None))
 		{
 			try
 			{
-				result = result
+				result
```

Comment:
Shouldn't the cancel token be passed to each verification?

Reply:
@soosr check latest commit. I've removed `IAsyncEnumerable` from all methods except `VerifyChangeAsync()`.

This eliminates all warnings of "this is async but you're not awaiting anything" and also makes the naming consistent: methods that don't have the "Async" suffix aren't really async